### ISSUE TITLE
tf-nightly-gpu builds from February are crashing on 'tf.keras model.fit()' for Nvidia RTX 3090; January builds performance is slower than 2080ti on : 2Layer Bidirectional LSTM + 3 Dense Layers

### ISSUE THREAD

**Issue by ALIANS-PRODUCTIONS** [tensorflow/tensorflow#47473]
**System information**
OS Windows 10 Pro : 1909 18363.1379
CUDA TOOLKIT : 11.1
CUDNN: 8.1.0
Latest GPU Driver 461.72

GPU : MSI RTX 3090 Gaming X Trio
CPU i7 3930k 32GB ram

using Anaconda
python 3.7.9
tf-nightly-gpu 2.5.0.dev20210114
tf-nightly 2.5.0.dev20210114

**Describe the problem**

=> The above config allows me to run my code LSTM code in tensor flow (extract below):
However if i create an environment in Anaconda by cloning the above and pip --upgrade to a tf-nightly build more recent, including todays build
of 2.5.0.dev20210228 ... when the code reaches tf.model.fit .. it crashes out with no error reporting at all.

The reason why i tried upgrading the tf-nightly build : currently the model training time is about the same as on my RTX 2080ti and sometimes even slower which i thought was down to issues with the tf-nightly january Build and Ampere.

**Provide the exact sequence of commands / steps that you executed before running into the problem**
The code can be run from CMD conda command prompt or from a terminal in Visual Studio Code and it crashes as soon as the model.fit line is reached.

**Any other info / logs**
code block:

model = tf.keras.models.Sequential([
#input shape equals term points in curve
tf.keras.layers.Bidirectional(tf.keras.layers.LSTM(32,return_sequences=True),input_shape=(n_days,n_cols)),
tf.keras.layers.Dropout(0.2),

tf.keras.layers.Bidirectional(tf.keras.layers.LSTM(32)),
tf.keras.layers.Dropout(0.2),

tf.keras.layers.Dense(64, activation='relu',kernel_initializer=initializer),
tf.keras.layers.BatchNormalization(),
tf.keras.layers.Dropout(0.2),
    
tf.keras.layers.Dense(64, activation='relu',kernel_initializer=initializer, bias_initializer=output_bias),
tf.keras.layers.BatchNormalization(),
tf.keras.layers.Dropout(0.2),

tf.keras.layers.Dense(t_cols, activation='sigmoid', bias_initializer=output_bias)])
#list of keras metrics to compute in model
METRICS = [
keras.metrics.BinaryAccuracy(name='binary_accuracy'),
keras.metrics.Precision(name='precision'),
keras.metrics.Recall(name='recall'),
keras.metrics.AUC(name='auc'),
]

model.compile(optimizer=Adam(learning_rate=lng_rate),
loss='binary_crossentropy',
metrics=METRICS)

history=model.fit(
X_train,
Y_train,
epochs=training_epochs,
batch_size=training_batch_size,
verbose=1,
class_weight=label_weights[0],
validation_data=(X_validation,Y_validation))

**Comment by ALIANS-PRODUCTIONS** [comment]
In addition to what i wrote above :  i have tested the issue with different sample codes and can replicate the error.
Attached is the jupyter notebook from Laurence Mooneys Tensorflow for Developer Course (from Coursera).
It runs absolutely fine on Google colab, on my tf-nighlty build from January (as written above) but crashes as soon as the code reaches the model.fit section  

[S+P_Week_3_Lesson_4_LSTM.zip](https://github.com/tensorflow/tensorflow/files/6062232/S%2BP_Week_3_Lesson_4_LSTM.zip)

**Comment by ALIANS-PRODUCTIONS** [comment]
Lastly to check that it wasnt my python environment in Anaconda that was corrupt. 
I have created a fresh new virtual env, python3.7, with tf-nightly-gpu 2.5.0.dev20210228, tf-nightly, tf-nightly-gpu 2.5.0.dev20210228, conda install pandas, numpy, scikit-learn and matplotlib : and still get the crash as described above when running the code on my 3090. I have a seperate pc with an RTX 2080ti and it has no issues.

**Comment by ALIANS-PRODUCTIONS** [comment]
I think the issue is with LSTMs in the current TF-nightly builds : they crash on a RTX 3090. A basic sequential model seems to run fine

**Comment by Saduf2019** [comment]
@ALIANS-PRODUCTIONS 
Can you please use the table version of tf 2.4 and let us know if you face the same issue.

Meanwhile please have a look at existing issue and let us know:
#46093, #44959, #46851

**Comment by ALIANS-PRODUCTIONS** [comment]
Hi, i downgraded to cuda 11.0 : kept everything else the same : same crash on the 3090 as before.
Next i kept cuda 11.0 and downgraded from tf-nightly-gpu and tf-nightly TO tf 2.4.1 : and it runs fine!
I then UPGRADED cuda back to 11.1 : keeping tf 2.4.1 : also runs fine.

So there is an issue with the current tf-nightly builds for 3090s.

I am happy to use tf 2.4.1 but will the fix for issue : multi-label class_weight be included in tf 2.4.1 ? 
>>Generalising class weights for all label ranks #46189

**Comment by Saduf2019** [comment]
@ALIANS-PRODUCTIONS 
Please create a new issue for this query, as this issue is related to build and install and cuda  11.1 is not a tested we suggest to use 11.0. As this is resolved, we request you to move this to closed status and create anew issue for the query.

**Comment by ALIANS-PRODUCTIONS** [comment]
> @ALIANS-PRODUCTIONS
> Please create a new issue for this query, as this issue is related to build and install and cuda 11.1 is not a tested we suggest to use 11.0. As this is resolved, we request you to move this to closed status and create anew issue for the query.


its not resolved. tf-nightly build from Feb crashes on ALL versions of cuda for 3090. 
Apologies if i am misunderstanding you here. Are you saying that for now, for 3090 users: they cannot use any of the recent tf-nightly builds ? and have to use tf 2.4.1?

**Comment by JohnStott** [comment]
Having the same frustrating issues!  

Using Cuda 11.0 with v2.4.1 is not the solution as the RTX 3090 has specific optimisations in cuda 11.2 which requires us to compile our own version of TF in order to benefit from the improved GPU speed.  It is not ideal having to pair this with v2.4.1, when we'd prefer tf-nightly.

@ALIANS-PRODUCTIONS - out of interest where do you look to see that the nightly job is failing (do you have a link?), just so I can keep checking on here to see if it eventually gets fixed?

**Comment by ALIANS-PRODUCTIONS** [comment]
Hi there , ,

1) I have found that current performance on TF 2.4.1 with CUDA 11.0 and 3090 : is about same and sometimes even slower than my 2080ti which was quite demoralising considering the price/potential. So completely agree with you about the future potential when everything is fully working. 

I dont have the familiarity with UNIX/ going the docker route which others are doing to use NVIDIAS TensorFlow builds (where they ARE seeing significant performance gains on the 3090 over previous gen, puget system reviews - lambda labs etc)  

Also using MIXED precision makes the code run almost twice as slow (both 2080ti and 3090) but Tensorflow docs do mention that the speed up is on special use case for very very large networks and you should take care in implementation. So i have turned it off. 

2) on my second comment you will find a zip file containing a notebook from the Tensorflow developer certification course. This shows a simple LSTM solution for a Timeseries generated in the notebook. The sheet works fine on colab, TF 2.4.1, but not TF nightly for the 3090 when it reaches model.fit. I cannot figure out where deep in the tensorflow code it crashes (I happily raise my hand up to declare my ignorance to the inner nuts and bolts) 
3) I have also noticed that a simple DENSE layer model will actually work fine.  so maybe its a KERAS specific issue but am blindly guessing here.

**Comment by sanatmpa1** [comment]
@ALIANS-PRODUCTIONS,

Can you try building the latest stable version of tensorflow i.e `2.6.0` and use this [guide](https://www.tensorflow.org/install/source_windows) for reference. Please update us if the issue still persists. Thanks!
